### PR TITLE
Feature check sbml success

### PR DESCRIPTION
**Pull request by jvanhoefer**
Adds additional checks when the SBML is created, if e.g. parameter/species creation was successful (i.e. if the id is a valid SBML identifier) and if the math can be parsed...

### CODE REVIEW

File: yaml2sbml/yaml2sbml.py [hunk 1]
```diff
@@ -411,21 +449,25 @@ def _create_species(model: sbml.Model, species_id: str, initial_amount: str):
...
     r = model.createRateRule()
-    r.setId('d/dt_' + species)
-    r.setVariable(species)
+    r.setId('d_dt_' + species_id)
+    r.setVariable(species_id)
     math_ast = sbml.parseL3Formula(formula)
+    if math_ast is None:
+        raise RuntimeError(f'Unable to generate the rate rule for the state '
+                           f'{species_id}, libsbml can not parse the right '
+                           f'hand side, given by {formula}).')
```
Comment (yannikschaelte) [inline yaml2sbml/yaml2sbml.py:470]:
```suggestion
                           f'{species_id}, libsbml can not parse the right-'
                           f'hand side, given by {formula}).')
```

File: yaml2sbml/yaml2sbml.py [hunk 2]
```diff
@@ -306,14 +319,26 @@ def _create_assignment(model: sbml.Model, assignment_id: str, formula: str):
...
-    assignment_rule.setMath(sbml.parseL3Formula(formula))
+
+    sbml_formula = sbml.parseL3Formula(formula)
+
+    if sbml_formula is not None:
+        assignment_rule.setMath(sbml.parseL3Formula(formula))
+    else:
+        raise RuntimeError(f'Unable to generate assignment for formula '
+                           f'{formula}, libsbml can not parse the given '
+                           f'expression.')
```
Comment (yannikschaelte) [inline yaml2sbml/yaml2sbml.py:341]:
```suggestion
        raise RuntimeError('Unable to generate assignment for formula '
                           f'{formula}, libsbml can not parse the given '
                           'expression.')
```

Comment (yannikschaelte) [reply yaml2sbml/yaml2sbml.py:341]: would also work, matter of style I guess

**Review by yannikschaelte — APPROVED** [review]

File: yaml2sbml/yaml2sbml.py (hunk 2 above)
Comment (jvanhoefer) [reply yaml2sbml/yaml2sbml.py:341]: I think I once had a style checker complain about this :D So slightly in favor of leaving it consistent :)

**Comment by jvanhoefer** [comment]
@dilpath There is a weird AMICI issue when importing AMICI in the notebook tests, that was not there yesterday (since then only the string of Error messages changed, but not the code in this PR). Do you have any idea, why this could be the case?

**Pull request merged**